### ISSUE TITLE
Limits on columns

### ISSUE THREAD

**Issue by Nyco** [nextcloud/deck#2008]
**Is your feature request related to a problem? Please describe.**
Deck is often advertised as a kanban style board, althgough it can be used in many more use cases.

One of the fundamental principles of kanban is the limits of tickets per column, that goes along with flow visualisation, as it helps the team manage and regulate the flow avoiding bottlenecks. Sometimes it is called "WIP limit" (Work In Progress).

**Describe the solution you'd like**
Option on each column, that can be accessed via the meatballs menu ("..." or horizontal three dots):
* No limit
* Soft limit (visual and textual alert, for example "Limit reached" in red)
* Hard limit (prevent card creation, suggest to create the card in the previous column if limit not reached yet)

**Describe alternatives you've considered**
Manually write a soft limit in plain text, in the column title, or as a first card (that would be differenciated and unmovable by agreement of team members).

**Additional context**

A visual example:
![Deck-limits](https://user-images.githubusercontent.com/25348/84197755-bff2b680-aaa2-11ea-929c-59b6e2788dfd.png)

* Soft limit in the column title: "Stuff in flow [limit: 4]"
* Soft limit in the first card: "Limit: 4 cards"

**Comment by JoergPalmer** [comment]
Would like to see this feature as well.

**Comment by juliushaertl** [comment]
@JoergPalmer Please use the [Github Reactions](https://github.com/blog/2119-add-reactions-to-pull-requests-issues-and-comments) feature instead of commenting that also need a specific feature. That way we can see your support without notifying all users subscribed to a given issue.

**Comment by juliushaertl** [comment]
I think a soft limit would be something we could implement with small effort, UI wise the Trello power up for list limits seems quite nice to me: https://trello.com/power-ups/5c2462c384ab8949b1724a20/list-limits

A hard limit would of course require a lot more handling also on the backend side and probably support in the android app, which I would actually like to avoid for an feature that might only be interesting to a partial of the user base.

Independent from the type of limit we add, we of course also need some new UI to manage settings per stack then.

**Comment by Nyco** [comment]
Yes, a soft limit seems to make sense given the constraints and the partial user base. The Trello power up UI seems to be quite intuitive and visual enough.

**Comment by linuxluser** [comment]
I would also like to add that a WIP limit is very essential to many Kanban methods. From popular [search hits](https://kanbanblog.com/explained/) about Kanban to [popular books](https://www.amazon.com/Project-Management-Kanban-Developer-Practices/dp/0735698953), WIP limit is deemed as the secret to success for Kanban.

I have not developed for Nextcloud yet and I do not know PHP but I'd be willing to help with this effort if you have something that I could work on. Please just point me in the right direction. This issue is important to me.

**Comment by Nyco** [comment]
* Jira: https://www.atlassian.com/agile/kanban/wip-limits
* GitLab: https://gitlab.com/gitlab-org/gitlab/-/issues/11403
* Redmine: https://www.redmineup.com/pages/help/agile/work-in-progress
* Kanbanize: https://kanbanize.com/fr/ressources/debuter-avec/limite-kanban-tec-travaux-en-cours (sorry, in French)

**Comment by wagnbeu0** [comment]
I would also like this feature which makes a kanban board more effective for Teams

**Comment by midnightfapper** [comment]
Please add WIP functionality!

**Comment by tomdavidson** [comment]
Kanban  (littles law) needs aggregate WIP limit for each in progress column.  Its funny having column limits for example we could limit a backlog but those items are not WIP.  Across all WIP columns we could have an agg of 10 but if one of those columns is 2, then the real pinch is 2 and not 10.

Ive never contrib to NextCloud but happy to help another with this one.

**Comment by tomdavidson** [comment]
I do think soft limits would satisfy 99% of the use cases. With soft limits I would like to have metrics of state changes & wip limit exceeded so that data would be ready available in our flow meeting.
